### PR TITLE
Fix MULE-8563. Add the posibility to define a name different from the fi...

### PR DESCRIPTION
**Pull request by pablocabrera85**
Add @JsonProperty handling to allow developer to use a different property names when adding extra authorization parameters

### CODE REVIEW

File: modules/devkit-support/src/main/java/org/mule/security/oauth/processor/BaseOAuth2AuthorizeMessageProcessor.java [hunk 1]
```diff
@@ -134,12 +136,16 @@ private void setState(Map<String, String> extraParameters, MuleEvent event)
 
                 try
                 {
+                    JsonProperty property=field.getAnnotation(JsonProperty.class);
```
Comment (marianogonzalez) [inline modules/devkit-support/src/main/java/org/mule/security/oauth/processor/BaseOAuth2AuthorizeMessageProcessor.java]: do not couple to Jackson. Define your own annotations for this

File: modules/devkit-support/src/main/java/org/mule/security/oauth/processor/BaseOAuth2AuthorizeMessageProcessor.java [hunk 2]
```diff
@@ -134,12 +136,16 @@ private void setState(Map<String, String> extraParameters, MuleEvent event)
...
+                    if(property!=null && StringUtils.isNotEmpty(property.value())){
```
Comment (marianogonzalez) [inline modules/devkit-support/src/main/java/org/mule/security/oauth/processor/BaseOAuth2AuthorizeMessageProcessor.java]: apply correct code formatting

File: modules/devkit-support/src/test/java/org/mule/security/oauth/processor/OAuth2AuthorizeMessageProcessorTestCase.java [hunk 3]
```diff
@@ -215,6 +222,9 @@ public String answer(InvocationOnMock invocation) throws Throwable
         private String customField;
         private String anotherCustomField;
 
+        @JsonProperty(annotatedPropertyName)
```
Comment (marianogonzalez) [inline modules/devkit-support/src/test/java/org/mule/security/oauth/processor/OAuth2AuthorizeMessageProcessorTestCase.java]: is devkit supposed to generate such a field on the MP if the connector is annotated somehow?

File: modules/devkit-support/src/main/java/org/mule/security/oauth/processor/BaseOAuth2AuthorizeMessageProcessor.java (hunk 1 above)
Comment (marianogonzalez) [inline modules/devkit-support/src/main/java/org/mule/security/oauth/processor/BaseOAuth2AuthorizeMessageProcessor.java]: I don't think this code belongs here. devkit-support module is annotations agnostic (actually, devkit annotations have compile retention and mule does not and will not depend on them).

The AuthorizationParameter are obtained by calling moduleObject.getDefaultUnauthorizedConnector().getAuthorizationParameters();

So, if what you want to specify aliases for them, then you should add code on the connector itself so that parameter.getName() returns the alias instead of the real name. This provides better compatibility between different connector and mule versions.

Notice that if you build a connector in which you use the alias functionality and deploy it on a mule version which doesn't have this improvement, then your alias would not work. If you do it on the connector, you'll have better portability.

In summary, I think that this is a devkit concern, not a mule one. If you agree, I'll close both this PR and the Jira issue

File: modules/devkit-support/src/main/java/org/mule/security/oauth/processor/BaseOAuth2AuthorizeMessageProcessor.java [hunk 4]
```diff
@@ -118,15 +119,16 @@ private void setState(Map<String, String> extraParameters, MuleEvent event)
         {
             for (AuthorizationParameter<?> parameter : params)
             {
+                String parameterName = parameter.getName();
                 Field field = null;
                 try
                 {
-                    field = this.getClass().getDeclaredField(parameter.getName());
+                    field = this.getClass().getDeclaredField(parameterName);
```
Comment (pablocabrera85) [inline modules/devkit-support/src/main/java/org/mule/security/oauth/processor/BaseOAuth2AuthorizeMessageProcessor.java]: This line prevents us from specifying an alias directly on DevKit. If the field I need to add on the map needs the name "Api-Key", java doesn't allow you to have an "Api-Key" field, since it is not a valid name. Get declared field will crash.

File: modules/devkit-support/src/main/java/org/mule/security/oauth/processor/BaseOAuth2AuthorizeMessageProcessor.java (hunk 1 above)
Comment (pablocabrera85) [inline modules/devkit-support/src/main/java/org/mule/security/oauth/processor/BaseOAuth2AuthorizeMessageProcessor.java]: We cannot use an alias on DevKit because this is using reflection to add the field from the class.

I'll add a new annotation to decouple if from Jackson. 

Using an annotation was one possible way of avoiding modifying org.mule.common.security.oauth.AuthorizationParameter class to add an alias/propertyName field

File: modules/devkit-support/src/test/java/org/mule/security/oauth/processor/OAuth2AuthorizeMessageProcessorTestCase.java (hunk 3 above)
Comment (pablocabrera85) [inline modules/devkit-support/src/test/java/org/mule/security/oauth/processor/OAuth2AuthorizeMessageProcessorTestCase.java]: Yes, there will be some changes required on DevKit, a branch with this changes was already coded.

File: modules/devkit-support/src/main/java/org/mule/security/oauth/processor/BaseOAuth2AuthorizeMessageProcessor.java (hunk 1 above)
Comment (marianogonzalez) [inline modules/devkit-support/src/main/java/org/mule/security/oauth/processor/BaseOAuth2AuthorizeMessageProcessor.java]: What you're proposing still doesn't seem like the best solution to me.

AuthorizationParameter is a DTO used to bridge the Devkit way of representing things (which are mainly annotations) against Mule's. Additionally, If you define a new annotation then it belongs on Devkit's codebase not mule's. If it's on devkit's, then mule cannot access it because there're dependency issues which prevents Mule from knowing devkit's jars.

I would simply add a getAlias() method to AuthorizationParameter. So, when the field is fetch you use the name, but when you build the extra parameters map which is used to build the request you use the alias. In this way the design remains consistent, in the sense that AuthorizationParameter remains a DTO and the two worlds remain separated.

**Comment by marianogonzalez** [comment]
Hello Pablo,

Any updates on this discussion? WIll you be sending other PR or should be close this one?

Thank you

**Comment by pablokraan** [comment]
Closing this PR as there were no updates in a long time
